mono/mini/mini-amd64.c

```diff
@@ -3095,6 +3098,8 @@ emit_call_body (MonoCompile *cfg, guint8 *code, MonoJumpInfoType patch_type, gco
 		else {
 			MonoJumpInfo *jinfo = NULL;
 
+			g_assert (call && data && call->fptr == data);
```

Comment:
Could pass `call` for `data`, but the strong typing is a little nice, and it might be too weird, to mixup/change the patch data type.

Reply:
Cleaned up this interface, looking better now.